### ISSUE TITLE
(BUG) August Smart Lock Pro doesn't lock or unlock

### ISSUE THREAD

**Issue by cladmonitor** [tonesto7/homebridge-smartthings-v2#53]
## Verify the following before opening an trouble issue

**Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!**

-  [ X] That _OAuth_ is Enabled for the SmartApp under the IDE.
-  [X ] The SmartApp and Device Handler are using the latest code available.
-  [X ] That Both the SmartApps and Device Handlers have been _Published for You_ in the IDE.

---
## About Your Setup
-  How many devices are detected?: 5
-  Mobile App Version(Not required):
-  SmartApp Version: 2.1.1
-  Device Handler Version: NA
-  Homebridge Version: 0.4.50
-  NodeJS Version: 10.12.0

## Expected Behavior
**Tell us what you think should be happening**
Upon sending a homebridge lock or unlock command the device should lock or unlock.

## Current Behavior
**What happens instead of the expected behavior?**
The command never reaches the device

## Steps to Reproduce (for bugs)
**Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. Include code to reproduce, if relevant**

1. Added device as "Other" or as "Switch"
2. Both devices are detected and show up in Homekit with no issue
3. Command sent to unlock produce the following event:
NOTICE: Sending Device Command: unlock | Value: Nothing | DeviceID: (d200b920-ec6f-4e0d-b166-c66e5d2fd8b5) | SendToLocalHub: (true)
NOTICE: Sending Device Command: lock | Value: Nothing | DeviceID: (d200b920-ec6f-4e0d-b166-c66e5d2fd8b5) | SendToLocalHub: (true)
4. If Device is unlocked manually the Homekit seems to see it as unlocked but cant do anything to lock it.

## Context
**How has this issue affected you? What are you trying to accomplish?
**Providing context helps us come up twith a solution that is most useful in the real world**

---
Please include a copy of any relevant log output to assist in tracking down the bug

#####Device Added as "Other" in Smart App Debug Log of HB
DEBUG: Initializing New Device (August Smart Lock Pro | d200b920-ec6f-4e0d-b166-c66e5d2fd8b5)
DEBUG: August Smart Lock Pro ' | ' 'lock'
GOOD: [CHARACTERISTIC (function() {
  Characteristic.call(this, 'Lock Current State', '0000001D-0000-1000-8000-0026BB765291');
  this.setProps({
    format: Characteristic.Formats.UINT8,
    maxValue: 3,
    minValue: 0,
    validValues: [0,1,2,3],
    perms: [Characteristic.Perms.READ, Characteristic.Perms.NOTIFY]
  });
  this.value = this.getDefaultValue();
}) GET] lock (August Smart Lock Pro) | LastUpdate: (Sat Jan 11 2020 18:49:14 GMT+0000 (Coordinated Universal Time)) | Value: (1)
GOOD: [CHARACTERISTIC (function() {
  Characteristic.call(this, 'Lock Target State', '0000001E-0000-1000-8000-0026BB765291');
  this.setProps({
    format: Characteristic.Formats.UINT8,
    maxValue: 1,
    minValue: 0,
    validValues: [0,1],
    perms: [Characteristic.Perms.READ, Characteristic.Perms.WRITE, Characteristic.Perms.NOTIFY]
  });
  this.value = this.getDefaultValue();
}) GET] lock (August Smart Lock Pro) | LastUpdate: (Sat Jan 11 2020 18:49:14 GMT+0000 (Coordinated Universal Time)) | Value: (1)
GOOD: [CHARACTERISTIC (function() {
  Characteristic.call(this, 'Lock Target State', '0000001E-0000-1000-8000-0026BB765291');
  this.setProps({
    format: Characteristic.Formats.UINT8,
    maxValue: 1,
    minValue: 0,
    validValues: [0,1],
    perms: [Characteristic.Perms.READ, Characteristic.Perms.WRITE, Characteristic.Perms.NOTIFY]
  });
  this.value = this.getDefaultValue();
}) GET] lock (August Smart Lock Pro) | LastUpdate: (Sat Jan 11 2020 18:49:14 GMT+0000 (Coordinated Universal Time)) | Value: (1)
DEBUG: August Smart Lock Pro ' | ' 'battery'
GOOD: [CHARACTERISTIC (function() {
  Characteristic.call(this, 'Battery Level', '00000068-0000-1000-8000-0026BB765291');
  this.setProps({
    format: Characteristic.Formats.UINT8,
    unit: Characteristic.Units.PERCENTAGE,
    maxValue: 100,
    minValue: 0,
    minStep: 1,
    perms: [Characteristic.Perms.READ, Characteristic.Perms.NOTIFY]
  });
  this.value = this.getDefaultValue();
}) GET] battery (August Smart Lock Pro) | LastUpdate: (Sat Jan 11 2020 18:49:14 GMT+0000 (Coordinated Universal Time)) | Value: (100)
GOOD: [CHARACTERISTIC (function() {
  Characteristic.call(this, 'Status Low Battery', '00000079-0000-1000-8000-0026BB765291');
  this.setProps({
    format: Characteristic.Formats.UINT8,
    maxValue: 1,
    minValue: 0,
    validValues: [0,1],
    perms: [Characteristic.Perms.READ, Characteristic.Perms.NOTIFY]
  });
  this.value = this.getDefaultValue();
}) GET] battery (August Smart Lock Pro) | LastUpdate: (Sat Jan 11 2020 18:49:14 GMT+0000 (Coordinated Universal Time)) | Value: (0)
GOOD: [CHARACTERISTIC (function() {
  Characteristic.call(this, 'Charging State', '0000008F-0000-1000-8000-0026BB765291');
  this.setProps({
    format: Characteristic.Formats.UINT8,
    maxValue: 2,
    minValue: 0,
    validValues: [0,1,2],
    perms: [Characteristic.Perms.READ, Characteristic.Perms.NOTIFY]
  });
  this.value = this.getDefaultValue();
}) GET] batteryStatus (August Smart Lock Pro) | LastUpdate: (Sat Jan 11 2020 18:49:14 GMT+0000 (Coordinated Universal Time)) | Value: (0)
INFO: Added Device: (August Smart Lock Pro | d200b920-ec6f-4e0d-b166-c66e5d2fd8b5)
ALERT: Total Initialization Time: (1 seconds)

######These are the only log entries when send an unlock or lock command (Debug is on)
NOTICE: Sending Device Command: unlock | Value: Nothing | DeviceID: (d200b920-ec6f-4e0d-b166-c66e5d2fd8b5) | SendToLocalHub: (true)

NOTICE: Sending Device Command: lock | Value: Nothing | DeviceID: (d200b920-ec6f-4e0d-b166-c66e5d2fd8b5) | SendToLocalHub: (true)

###### Device Data - Connected as "Other"
{
    "name": "August Smart Lock Pro",
    "basename": "August Smart Lock Pro",
    "deviceid": "d200b920-ec6f-4e0d-b166-c66e5d2fd8b5",
    "status": "ONLINE",
    "manufacturer": "Unknown",
    "model": "Z-Wave Lock Without Codes",
    "deviceNetworkId": "07",
    "lastActivity": "2020-01-11T18:55:08+0000",
    "capabilities": [
        "Actuator",
        "Battery",
        "Configuration",
        "Health Check",
        "Lock",
        "Refresh",
        "Sensor"
    ],
    "commands": [
        "configure",
        "lock",
        "ping",
        "refresh",
        "unlock"
    ],
    "customflags": [ ],
    "attributes": {
        "DeviceWatch-DeviceStatus": null,
        "DeviceWatch-Enroll": null,
        "battery": 100,
        "checkInterval": 3600,
        "healthStatus": null,
        "lock": "unlocked with timeout"
    },
    "eventHistory": [
        "Sat Jan 11 18:55:08 UTC 2020 | [lock] | (unlocked with timeout)", <--- Manually unlocked
        "Sat Jan 11 18:40:04 UTC 2020 | [lock] | (locked)", <-- Unknown (maybe status req)
        "Sat Jan 11 18:40:04 UTC 2020 | [lock] | (locked)", <-- Unknown (maybe status req)
        "Sat Jan 11 18:40:00 UTC 2020 | [null] | (lock)", <-- homekit command
        "Sat Jan 11 18:39:58 UTC 2020 | [null] | (unlock)", <-- homekit command
        "Sat Jan 11 18:32:57 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:52 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:51 UTC 2020 | [null] | (unlock)",
        "Sat Jan 11 18:32:47 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:46 UTC 2020 | [null] | (lock)",
        "Sat Jan 11 18:32:41 UTC 2020 | [null] | (unlock)",
        "Sat Jan 11 18:32:31 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:26 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:24 UTC 2020 | [null] | (lock)",
        "Sat Jan 11 18:32:24 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:22 UTC 2020 | [null] | (unlock)",
        "Sat Jan 11 18:32:17 UTC 2020 | [null] | (lock)",
        "Sat Jan 11 18:32:06 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:03 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 18:32:00 UTC 2020 | [null] | (unlock)"
    ]
}

**Comment by cladmonitor** [comment]
So if you manually unlock and lock the device seems to work for one iteration (unlock though homekit and then relock through homekit) but then not again.

Hope all this helps.

{
    "name": "August Smart Lock Pro",
    "basename": "August Smart Lock Pro",
    "deviceid": "d200b920-ec6f-4e0d-b166-c66e5d2fd8b5",
    "status": "ONLINE",
    "manufacturer": "Unknown",
    "model": "Z-Wave Lock Without Codes",
    "deviceNetworkId": "07",
    "lastActivity": "2020-01-11T19:00:45+0000",
    "capabilities": [
        "Actuator",
        "Battery",
        "Configuration",
        "Health Check",
        "Lock",
        "Refresh",
        "Sensor"
    ],
    "commands": [
        "configure",
        "lock",
        "ping",
        "refresh",
        "unlock"
    ],
    "customflags": [  ],
    "attributes": {
        "DeviceWatch-DeviceStatus": null,
        "DeviceWatch-Enroll": null,
        "battery": 100,
        "checkInterval": 3600,
        "healthStatus": null,
        "lock": "locked"
    },
    "eventHistory": [
        "Sat Jan 11 19:00:45 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:44 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:44 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:41 UTC 2020 | [lock] | (unlocked)",
        "Sat Jan 11 19:00:41 UTC 2020 | [lock] | (unlocked with timeout)",
        "Sat Jan 11 19:00:40 UTC 2020 | [null] | (lock)",
        "Sat Jan 11 19:00:34 UTC 2020 | [null] | (unlock)",
        "Sat Jan 11 19:00:29 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:29 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:26 UTC 2020 | [lock] | (unlocked)",
        "Sat Jan 11 19:00:26 UTC 2020 | [lock] | (unlocked with timeout)",
        "Sat Jan 11 19:00:26 UTC 2020 | [lock] | (unlocked)",
        "Sat Jan 11 19:00:25 UTC 2020 | [lock] | (unlocked with timeout)",
        "Sat Jan 11 19:00:25 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:24 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:21 UTC 2020 | [lock] | (unlocked)",
        "Sat Jan 11 19:00:21 UTC 2020 | [lock] | (unlocked with timeout)",
        "Sat Jan 11 19:00:16 UTC 2020 | [lock] | (locked)",
        "Sat Jan 11 19:00:10 UTC 2020 | [null] | (lock)",
        "Sat Jan 11 19:00:04 UTC 2020 | [lock] | (locked)"
    ]
}

**Comment by tonesto7** [comment]
@cladmonitor  I don't see anything wrong with that logs from the plugin.  
What are you seeing in the logs ST side? (You do not need to enable debug logs to see it)

**Comment by cladmonitor** [comment]
It’s interesting I tested in ST prior to making the bug report and it worked flawless. I tried now to look at event history and it’s certainly not behaving correctly.

I will say August Locks are pretty terrible, they have issues all over the app with inconsistent workflows missing documentation and strange hardware behavioral ticks.

I’ll just blame them and move on, I’d rather use ZWave but for now I can use the Wifi Bridge and native HomeKit (you loose battery info which really sucks).

Sorry to send you on any wild goose chase and appreciated all your hard work!!
